build/source/driver/multi_driver.f90

```diff
@@ -384,130 +433,139 @@ real(dp),parameter        :: safety = 0.01_dp               ! safety factor to e
-
-end do  ! (looping through HRUs)
-
-! allocate space for the upslope area
-allocate(upArea(nHRU),stat=err); call handle_err(err,'problem allocating space for upArea')
+ ! loop through local HRUs
+ do iHRU=1,gru_struc(iGRU)%hruCount
+
+  ! check that the parameters are consistent
+  call paramCheck(mparStruct%gru(iGRU)%hru(iHRU)%var,err,message); call handle_err(err,message)
+  
+  ! calculate a look-up table for the temperature-enthalpy conversion 
+  call E2T_lookup(mparStruct%gru(iGRU)%hru(iHRU)%var,err,message); call handle_err(err,message)
+ 
+  ! read description of model initial conditions -- also initializes model structure components
+  ! NOTE: at this stage the same initial conditions are used for all HRUs -- need to modify
+  call read_icond(gru_struc(iGRU)%hruInfo(iHRU)%nSnow, & ! number of snow layers
+                  gru_struc(iGRU)%hruInfo(iHRU)%nSoil, & ! number of soil layers
+                  mparStruct%gru(iGRU)%hru(iHRU)%var,  & ! vector of model parameters
+                  indxStruct%gru(iGRU)%hru(iHRU),      & ! data structure of model indices
+                  progStruct%gru(iGRU)%hru(iHRU),      & ! model prognostic (state) variables
+                  err,message)                           ! error control
+  call handle_err(err,message)
+  
+  ! re-calculate height of each layer
+  call calcHeight(&
+                  ! input/output: data structures
+                  indxStruct%gru(iGRU)%hru(iHRU),   & ! intent(in): layer type
+                  progStruct%gru(iGRU)%hru(iHRU),   & ! intent(inout): model prognostic (state) variables for a local HRU
+                  ! output: error control
+                  err,message); call handle_err(err,message)
+  
+  ! calculate vertical distribution of root density
+  call rootDensty(mparStruct%gru(iGRU)%hru(iHRU)%var,& ! vector of model parameters
+                  indxStruct%gru(iGRU)%hru(iHRU),    & ! data structure of model indices
+                  progStruct%gru(iGRU)%hru(iHRU),    & ! data structure of model prognostic (state) variables
+                  diagStruct%gru(iGRU)%hru(iHRU),    & ! data structure of model diagnostic variables
+                  err,message)                         ! error control
+  call handle_err(err,message) 
+  
+  ! calculate saturated hydraulic conductivity in each soil layer
+  call satHydCond(mparStruct%gru(iGRU)%hru(iHRU)%var,& ! vector of model parameters
+                  indxStruct%gru(iGRU)%hru(iHRU),    & ! data structure of model indices
+                  progStruct%gru(iGRU)%hru(iHRU),    & ! data structure of model prognostic (state) variables
+                  fluxStruct%gru(iGRU)%hru(iHRU),    & ! data structure of model fluxes 
+                  err,message)                         ! error control
+  call handle_err(err,message)
+  
+  ! calculate "short-cut" variables such as volumetric heat capacity
+  call v_shortcut(mparStruct%gru(iGRU)%hru(iHRU)%var,& ! vector of model parameters
```

Comment:
Since we can take many shortcuts (and probably will), provide a less generic name.